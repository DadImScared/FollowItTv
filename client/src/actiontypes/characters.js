
export const ADD_CHARACTERS = 'characters/ADD_CHARACTERS';
