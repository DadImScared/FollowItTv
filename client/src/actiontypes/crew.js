
export const ADD_CREW = 'crew/ADD_CREW';
