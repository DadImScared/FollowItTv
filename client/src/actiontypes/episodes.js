
export const ADD_EPISODES = 'episodes/ADD_EPISODES';
