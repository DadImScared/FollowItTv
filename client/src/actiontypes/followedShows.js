
export const FOLLOW_SHOW = 'followedShows/FOLLOW_SHOW';
export const UNFOLLOW_SHOW = 'followedShows/UNFOLLOW_SHOW';
export const ADD_DAY = 'followedShows/ADD_DAY';
