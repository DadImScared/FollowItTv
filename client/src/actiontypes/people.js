
export const ADD_PEOPLE = 'people/ADD_PEOPLE';
