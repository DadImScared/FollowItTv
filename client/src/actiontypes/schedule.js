
export const ADD_SCHEDULE = 'schedule/ADD_SCHEDULE';
