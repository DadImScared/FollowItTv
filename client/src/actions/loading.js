
import { LOADING_START, LOADING_END } from '../actiontypes/loading';

export const loadingStart = (requestId) => {
  return {
    type: LOADING_START,
    requestId
  };
};

export const loadingEnd = (requestId) => {
  return {
    type: LOADING_END,
    requestId
  };
};
