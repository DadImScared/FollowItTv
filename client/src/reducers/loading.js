
import { LOADING_START, LOADING_END } from '../actiontypes/loading';

function loading(state = {}, action) {
  switch(action.type) {
    case LOADING_START:
      return {
        ...state,
        [action.requestId]: true
      };
    case LOADING_END:
      return {
        ...state,
        [action.requestId]: false
      };
    default:
      return state;
  }
}

export default loading;
