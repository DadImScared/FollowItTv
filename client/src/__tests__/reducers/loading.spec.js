
import { LOADING_START, LOADING_END } from '../../actiontypes/loading';
import reducer from '../../reducers/loading';

describe('loading reducer', () => {
  it('should add loading state', () => {
    expect(
      reducer({}, {
        type: LOADING_START,
        requestId: 'REQUEST_ID'
      })
    ).toMatchSnapshot();
  });

  it('should set loading state to false', () => {
    expect(
      reducer({ 'REQUEST_ID': true }, {
        type: LOADING_END,
        requestId: 'REQUEST_ID'
      })
    ).toMatchSnapshot();
  });
});
