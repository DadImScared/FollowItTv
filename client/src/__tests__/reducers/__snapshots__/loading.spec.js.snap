// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading reducer should add loading state 1`] = `
Object {
  "REQUEST_ID": true,
}
`;

exports[`loading reducer should set loading state to false 1`] = `
Object {
  "REQUEST_ID": false,
}
`;
